libs/providers/ofrep/src/lib/ofrep-provider.ts

```diff
@@ -0,0 +1,80 @@
+import { EvaluationContext, JsonValue, Provider, ResolutionDetails, TypeMismatchError } from '@openfeature/server-sdk';
+import {
+  EvaluationFlagValue,
+  handleEvaluationError,
+  OFREPApi,
+  OFREPApiEvaluationResult,
+  OFREPProviderBaseOptions,
+  toRequestOptions,
+  toResolutionDetails,
+} from '@openfeature/ofrep-core';
+
+export type OFREPProviderOptions = OFREPProviderBaseOptions;
+
+export class OFREPProvider implements Provider {
+  private ofrepApi: OFREPApi;
+
+  readonly runsOn = 'server';
+  readonly metadata = {
+    name: OFREPProvider.name,
```

Comment:
I think we should define what the provider name should be before merging this. Perhaps we could use the full name like this "OpenFeature Remote Evaluation Protocol Provider". We talked about avoiding using provider in the name in the past but I think it's valuable to include in this case.

FYI @toddbaert @thomaspoignant @lukas-reining

Reply:
I almost made the same comment, so it's probably a good Idea. We should decide on a canonical name, probably without "provider".

Reply:
You mean we set the following? 

```ts
  readonly metadata = {
    name: "OpenFeature Remote Evaluation Protocol Provider",
  }
```

Reply:
I think we will have to revisit that when we will have the metadata endpoint.

Reply:
> You mean we set the following?

Yes, exactly. That's what we show in telemetry, so it should be user-friendly and consistent across implementations.

Reply:
Okay I set it to that, is this fine for all?
@beeme1mr @toddbaert @thomaspoignant

Reply:
I would add the server in it.
Something like :
```
  readonly metadata = {
    name: "OpenFeature Remote Evaluation Protocol Server Provider",
  }
```